[#9] Criaçao do documento de metodologia (#13)
UnBArqDsw2020-2/2020.2_G2_Encare: docs/metodologia.md

# Metodologia:

## Introdução:

Segundo o dicionário michelis [1], metodologia pode ser entendida como um conjunto de regras e procedimentos para a realização de uma pesquisa. Na área de software a metodologia tem sentido semelhante, sendo também um conjunto de regras e procedimentos que regem um projeto.

> A fim de poder economizar tempo e poder se mais produtivo a equipe definiu as metodologias que serão usadas para o seguimento do projeto. Este documento tem por objetivo explicar quais são as metodologias escolhidas e comos elas inteferem no modo de trabalhar da equipe.

Reviewer: ```suggestion
A fim de poder economizar tempo e ser mais produtivo, a equipe definiu as metodologias que serão usadas para o seguimento do projeto. Este documento tem por objetivo explicar quais são as metodologias escolhidas e como elas interferem no modo de trabalhar da equipe.
```

## Metodologias escolhidas:
### Scrum

> O Scrum é uma metodologia ágil para gestão e planejamento de projetos de software [2] e é a base da escolha metodológica de trabalho da equipe. Os principais prodcedimentos a serem adotados pela equipe do Scrum são:

Reviewer: ```suggestion
O Scrum é uma metodologia ágil para gestão e planejamento de projetos de software [2] e é a base da escolha metodológica de trabalho da equipe. Os principais procedimentos a serem adotados pela equipe do Scrum são:
```

>  - **Daily meeting**: Reuniões diárias que tem por objetivo resumir o que a pessoa fez no dia e falar oq ue ela planeja fazer no próximo, além de citar quais são os seus impedimentos. Isso ajuda a toda a equipe estar ciente do andamento do projeto como um todo, além de poder identificar dependências.

Reviewer: ```suggestion
 - **Daily meeting**: Reuniões diárias que tem por objetivo resumir o que a pessoa fez no dia e falar o que ela planeja fazer no próximo, além de citar quais são os seus impedimentos. Isso ajuda a toda a equipe estar ciente do andamento do projeto como um todo, além de poder identificar dependências.
```

>  - **Sprints**: Sprints são ciclos que serão semanais - iniciando e terminando as terças, observando dias de entregas também.

Reviewer: ```suggestion
 - **Sprints**: Sprints são ciclos que serão semanais - iniciando e terminando às terças, observando dias de entregas também.
```

 - **Programação em pares**: É a programção em duplas em um unico computador, o que melhora o desenvolvimento em vários aspectos, como a descoberta de falhas. Além disso contribui possitivamente para o trabalho em equipe.
### Kanban
O Kanban é um sistema ágil e visual de gestão de tarefas dividindo as tarefas em colunas, que representam o status das tarefas inseridas em cada coluna. [4]
## Ferramentas usadas pela equipe:
### Telegram
É um aplicativo de messagens que serve como principal meio de comunicação de toda a equipe. Por ele que ocorre as notificações sobre o projeto, além de ocorrer as dailies de forma assíncrona.
### Google drive
É uma ferramenta de hospedagem, compartilhamento e desenvovimento de arquivos e é usado para hospedar as atas das reuniões, além de outros arquivos.
### Github
É a ferramenta de versionamento e de hospedagem do código e documentos do projeto.
### Github Pages - Mkdocs

> O github pages é uma página da web onde é aprensentada os documentos e outros arquivos. Já o mkdocs é uma ferramenta que facilita o uso de arquivos em markdown no github pages.

Reviewer: ```suggestion
O github pages é uma página da web onde é apresentada os documentos e outros arquivos. Já o mkdocs é uma ferramenta que facilita o uso de arquivos em markdown no github pages.
```

 - **Sprints**: Sprints são ciclos que serão semanais - iniciando e terminando às terças, observando dias de entregas também.
 - **Plannig**: É uma reunião semanal que acontece no início de cada sprint e nela ocorre o planejamento da sprint e divisão das tarefas.
 - **Product Backlog**: É uma lista de tarefas e funcionalidades que serão implementadas futuramente.
 - **Sprint Review**: É uma reunião que tem por objetivo revisar tudo o que foi feito na sprint passada, destacando pontos positivos e negativos. Na equipe essa reunião acontece antes de começar a planning.

### XP

O XP (extreme programming) é uma metodologia focada no desenvolvimento de software que possui valores e princípios que são fundamentados principalmente por um conjunto de práticas. Além disso é uma metodologia leve que pode ser adotada por diferentes níveis de programadores. A principal caracteristica do XP adotada pela equipe é a programação em pares.[3]

>  - **Programação em pares**: É a programção em duplas em um unico computador, o que melhora o desenvolvimento em vários aspectos, como a descoberta de falhas. Além disso contribui possitivamente para o trabalho em equipe.

Reviewer: ```suggestion
 - **Programação em pares**: É a programação em duplas em um único computador, o que melhora o desenvolvimento em vários aspectos, como a descoberta de falhas. Além disso contribui positivamente para o trabalho em equipe.
```

### Kanban

O Kanban é um sistema ágil e visual de gestão de tarefas dividindo-as em colunas, que representam o status das tarefas inseridas em cada coluna. [4]

## Ferramentas usadas pela equipe:

### Telegram

> É um aplicativo de messagens que serve como principal meio de comunicação de toda a equipe. Por ele que ocorre as notificações sobre o projeto, além das dailies de forma assíncrona.

Reviewer: ```suggestion
É um aplicativo de mensagens que serve como principal meio de comunicação de toda a equipe. Por ele que ocorre as notificações sobre o projeto, além das dailies de forma assíncrona.
```
